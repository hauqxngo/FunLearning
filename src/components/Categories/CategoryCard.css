.category-card {
  box-shadow: rgba(149, 157, 165, 0.2) 0px 8px 24px;
  transition: all 0.3s ease-out;
}

.category-card:hover {
  font-size: 20px;
  box-shadow: rgba(0, 0, 0, 0.16) 0px 10px 36px 0px,
    rgba(0, 0, 0, 0.06) 0px 0px 0px 1px;
}

.category-card:hover h5 {
  font-weight: bold;
}
