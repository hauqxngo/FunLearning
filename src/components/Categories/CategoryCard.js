import React from "react";
import { Link } from "react-router-dom";
import { Card, CardBody, CardTitle, CardText } from "reactstrap";
import "./CategoryCard.css";

/** Show info of a category on the Category List
 *
 * CategoryList -> CategoryCard
 */

const CategoryCard = ({ name, description, handle }) => {
  return (
    <div>
      <Link className="text-decoration-none" to={`/categories/${handle}`}>
        <Card body className="secondary mx-5 my-4 category-card" outline>
          <CardBody>
            <CardTitle className="text-success" tag="h5">
              {name}
            </CardTitle>
            <CardText className="text-dark">{description}</CardText>
          </CardBody>
        </Card>
      </Link>
    </div>
  );
};

export default CategoryCard;
