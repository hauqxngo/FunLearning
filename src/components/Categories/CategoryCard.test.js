import React from "react";
import { render } from "@testing-library/react";
import CategoryCard from "./CategoryCard";
import { MemoryRouter } from "react-router";

it("matches snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <CategoryCard
        handle="alphabet"
        name="Alphabet"
        description="A is for Apple. B is for Bus. C is for Cat."
      />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
