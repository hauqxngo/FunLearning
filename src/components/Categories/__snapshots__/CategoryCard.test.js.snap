// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <a
      class="text-decoration-none"
      href="/categories/alphabet"
    >
      <div
        class="secondary mx-5 my-4 category-card card card-body"
      >
        <div
          class="card-body"
        >
          <h5
            class="text-success card-title"
          >
            Alphabet
          </h5>
          <p
            class="text-dark card-text"
          >
            A is for Apple. B is for Bus. C is for Cat.
          </p>
        </div>
      </div>
    </a>
  </div>
</DocumentFragment>
`;
